instrumentation/metrics/jvm/src/main/scala/org/typelevel/otel4s/instrumentation/ce/IORuntimeMetricsPlatform.scala

```diff
@@ -0,0 +1,367 @@
+/*
+ * Copyright 2022 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.typelevel.otel4s.instrumentation.ce
+
+import cats.effect.Resource
+import cats.effect.Sync
+import cats.effect.unsafe.metrics.{IORuntimeMetrics => CatsIORuntimeMetrics}
+import cats.effect.unsafe.metrics.WorkStealingPoolMetrics
+import cats.syntax.applicative._
+import cats.syntax.flatMap._
+import cats.syntax.foldable._
+import cats.syntax.functor._
+import org.typelevel.otel4s.Attribute
+import org.typelevel.otel4s.Attributes
+import org.typelevel.otel4s.metrics.Meter
+import org.typelevel.otel4s.metrics.MeterProvider
+
+private[ce] trait IORuntimeMetricsPlatform {
+  self: IORuntimeMetrics.type =>
+
+  sealed trait Config {
+    def computeMetrics: Boolean
+    def computeMetricsAttributes: Attributes
+    def localQueueMetrics: Boolean
+    def localQueueMetricsAttributes: Attributes
+    def cpuStarvationMetrics: Boolean
+    def cpuStarvationMetricsAttributes: Attributes
+
+    def withComputeMetrics: Config
+    def withoutComputeMetrics: Config
+    def withComputeMetricsAttributes(attributes: Attributes): Config
```

Comment:
The config feels verbose. On the other hand, we can easily add support for new metrics (e.g., https://github.com/typelevel/cats-effect/pull/4187) in a binary-compatible way.